### PR TITLE
implement load test scripts for vrf and rr

### PR DESCRIPTION
**Pull request by Intizar-T**
# Description

- Load test scripts for vrf and rr in `inspector` directory.
- Update readme accordingly

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.

## Deployment

- [ ] Should publish npm package
- [ ] Should publish Docker image

### CODE REVIEW

File: inspector/inspector-task.ts
Comment (nick-bisonai) [inline inspector/inspector-task.ts:1]: it'd nice to check if values returned from vrf and rr after load test are also as expected, VRF returning random words and RR returning the actual value in url endpoint. if your intention was to just focus on blocks processing status, I won't insist

File: inspector/inspector-task.ts [hunk 1]
```diff
@@ -71,3 +74,172 @@ task("inspect", "Main task")
...
+        consumerContractName = "InspectorConsumer";
+        break;
+      case "vrf":
+        consumerContractName = "LoadTestVRFConsumer";
+        break;
+      case "rr":
+        consumerContractName = "LoadTestRRConsumer";
+        break;
+      default:
+        consumerContractName = "InspectorConsumer";
```
Comment (nick-bisonai) [inline inspector/inspector-task.ts:227]: as I suggested before, it'd be nice if we don't have to deploy multiple contracts to run inspector scripts

**Review by nick-bisonai — APPROVED** [review]
lgtm! left few comments

File: inspector/contracts/LoadTestRRConsumer.sol [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    }
+
+    function clear() public onlyOwner {
+        for (uint256 i = 0; i < allRequestIds.length; i++) {
+            delete requestBlockNumbers[allRequestIds[i]];
+        }
+
+        delete blockRecords;
+        delete allRequestIds;
+    }
```
[the remark this answers is not in the document]
Comment (nick-bisonai) [reply inspector/contracts/LoadTestRRConsumer.sol:74]: easiest approach i can think of is just clearing on request call 😂

[the remark this answers is not in the document]
Comment (nick-bisonai) [reply inspector/contracts/LoadTestRRConsumer.sol:74]: well this is optional suggestion so I won't insist

File: inspector/contracts/LoadTestRRConsumer.sol [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    function fulfillDataRequest(uint256 requestId, uint128 /* response */) internal override {
+        blockRecords.push(block.number - requestBlockNumbers[requestId]);
+    }
+
+    function fulfillRandomWords(
+        uint256 /* requestId */,
+        uint256[] memory  /* randomWords */
+    ) internal override {
+        // pass
+    }
```
[the remark this answers is not in the document]
Comment (Intizar-T) [reply inspector/contracts/LoadTestRRConsumer.sol:61]: compilation was giving me an error saying function definition is not provided. Seems like it expects the function decleratation

**Comment by Intizar-T** [comment]
This is an initial working prototype. Will merge this for now and create a new issue for myself to followup in the future (this is a non urgent feature)

**Pull request merged**